bundles/org.openhab.binding.elgatokeylight/src/main/java/org/openhab/binding/elgatokeylight/internal/ElgatoKeyLightHandler.java

```diff
@@ -0,0 +1,181 @@
+                    int brightness = LightStateConverter.toAdjustedBrightness((IncreaseDecreaseType) command,
+                            lightStatus.brightness);
+                    boolean on = brightness > 0;
+                    lightStatus = light.switchTo(on, brightness);
+                }
+                updateState(lightStatus);
+                break;
+
+            case CHANNEL_COLOR_TEMPREATURE:
+                if (command instanceof PercentType) {
+                    int temperature = LightStateConverter.toColorTemperatureLightState((PercentType) command);
+                    lightStatus = light.writeTemperature(temperature);
+                } else if (command instanceof OnOffType) {
+                    boolean on = OnOffType.ON.equals(command);
+                    lightStatus = light.switchTo(on, lightStatus.brightness > 0 ? lightStatus.brightness : 100);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    int temperature = LightStateConverter.toAdjustedColorTemp((IncreaseDecreaseType) command,
+                            lightStatus.brightness);
+                    lightStatus = light.writeTemperature(temperature);
+                }
+                updateState(lightStatus);
+                break;
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            return;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(ElgatoKeyLightConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily and let the background task
+        // decide for the real status.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            if ((config.host == null) || config.host.isEmpty()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Host not set!");
+                return;
+            }
+            if (config.port == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Port not set!");
+                return;
+            }
+            try {
+                light = new ElgatoKeyLight(config.host, config.port);
+                LightStatus lightStatus = light.readLightStatus();
+                if (lightStatus != null) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "No status returned querying device!");
+                }
+            } catch (Exception e) {
+                LOG.debug("Exception reading light status '{}'", config.host, e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Exception reading status: " + e.getMessage());
```

Comment:
The framework logs the message already. So, the log message could be removed. Logging the stack trace seems a bit unreasonable, when the network fails.